src/seedu/addressbook/logic/Logic.java

```diff
@@ -85,7 +85,7 @@ public CommandResult execute(String userCommandText) throws Exception {
     private CommandResult execute(Command command) throws Exception {
         command.setData(addressBook, lastShownList);
         CommandResult result = command.execute();
-        storage.save(addressBook);
+        if(command.isMutating()) storage.save(addressBook);
```

Comment:
- Coding standard violation detected; please check what our coding standards have to tell about conditional statements with a single line of code.
- Need to update the header comment of this `execute` method in line with the changed behaviour due to the newly added `isMutating` method.